spec/models/asset_spec.rb

```diff
@@ -198,6 +199,79 @@
 
       asset.scanned_clean!
     end
+
+    context 'when asset is already clean' do
+      let(:state) { 'clean' }
+
+      it 'does not allow the state transition' do
+        expect { asset.scanned_clean! }
+          .to raise_error(StateMachines::InvalidTransition)
+      end
+    end
+
+    context 'when asset is already infected' do
+      let(:state) { 'infected' }
+
+      it 'does not allow the state transition' do
+        expect { asset.scanned_clean! }
+          .to raise_error(StateMachines::InvalidTransition)
+      end
+    end
+
+    context 'when asset is already uploaded' do
+      let(:state) { 'uploaded' }
+
+      # before do
+      #   asset.file = load_fixture_file('asset.png')
+      # end
+
+      it 'does not allow the state transition' do
+        expect { asset.scanned_clean! }
+          .to raise_error(StateMachines::InvalidTransition)
+      end
+    end
+  end
+
+  describe 'when an asset is marked as infected' do
```

Comment:
There doesn't appear to be a test within this `describe` block that asserts that the asset is transitioned to the `infected` state. Is that tested elsewhere?

Reply:
@chrisroos:

> Is that tested elsewhere?

No, it doesn't look like it. I guess I was focussing on testing the effects of the extra constraints. However, I've now added an example in both this and the 'when an asset is marked as clean' `describe` block for completeness.